tests/src/Integration/Engine/AnnotationProcessingTest.php

```diff
@@ -23,11 +23,46 @@ class AnnotationProcessingTest extends RulesIntegrationTestBase {
   public function setUp() {
     parent::setUp();
     $this->enableModule('user');
+    // Some of our plugins assume sessions exist:
     $session_manager = $this->prophesize(SessionManagerInterface::class);
     $this->container->set('session_manager', $session_manager->reveal());
   }
 
   /**
+   * Make sure @ Translation annotations do not leak out into the wild.
+   */
+  public function testTranslationSquelching() {
+    // Get a sample Rules plugin.
+    $plugin = $this->conditionManager->createInstance('rules_list_contains');
+    $context = $plugin->getContext('list');
+    $definition = $context->getContextDefinition();
+
+    // These can reasonable be string or TranslatableMarkup, but never
+    // a Translation object.
+    $label = $definition->getLabel();
+    $description = $definition->getDescription();
+
+    $valid = !(is_object($label) && ($label instanceof Translation));
+    $this->assertTrue($valid, 'Label is not a Translation object');
```

Comment:
@klausi  Let me think... it's valid if it's a string, or if it's TranslableMarkup.  So actually, the test is "it's not a Translation object".  I'm probably testing is_object($label) on the assumption that the instanceof operator won't be happy if I pass it a non-object. Which might not be a valid assumption, BTW.

So, no, we can't use is_string().  But there may be a better test than what I'm using here, nonetheless.